provider/source/src/units/data.rs

```diff
@@ -126,17 +104,25 @@ impl crate::IterableDataProviderCached<UnitsDisplayNameV1Marker> for SourceDataP
         ) -> Result<(), DataError> {
             let quantities = length_patterns
                 .keys()
-                // TODO: remove this filter once we are supporting all the units categories.
-                // NOTE:
-                //  if this filter is removed, we have to add a filter to remove all the prefixes.
+                .filter(|&key| {
+                    !key.starts_with(|c: char| c.is_ascii_digit())
+                        && !["per", "times", "power"]
+                            .iter()
+                            .any(|&prefix| key.starts_with(prefix))
+                })
                 .filter_map(|key| {
+                    // In order to reduce the number of units in the test data,
+                    // we only test the length-* and duration-* units.
+                    #[cfg(test)]
```

Comment:
length + duration are 127 units, resulting in 2044 json files. Let's pick one or two concrete units here to reduce that number to 10-20 files.

Reply:
https://github.com/unicode-org/icu4x/issues/5226

Reply:
please do this now

Reply:
done here: https://github.com/unicode-org/icu4x/pull/5238

It is better to have the PRs lean and to the point for easier review. For example, in the current PR, if there is no change in the test data files, it is evidence that we have populated the baked data without affecting the test data. Similarly, in the other PR (https://github.com/unicode-org/icu4x/pull/5238), when it is only reducing the test data, the lack of effect on the baked data is evidence that we are heading in the right direction and only reducing the test data.